### PR TITLE
backend/fix : fixed showing latest driver linked with vehicle

### PR DESCRIPTION
**Pull request by AshishSaini25**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

fixed showing latest driver linked with vehicle

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/dev/migrations/provider-dashboard/0022-fleet-owner-access.sql [hunk 1]
```diff
@@ -3,4 +3,8 @@ insert into atlas_bpp_dashboard.access_matrix (id, role_id, api_entity, user_acc
...
-insert into atlas_bpp_dashboard.access_matrix (id, role_id, api_entity, user_access_type, created_at, updated_at, user_action_type) (select atlas_bpp_dashboard.uuid_generate_v4() as id,id as role_id,'FLEET' as api_entity, 'USER_FULL_ACCESS' as user_access_type, now() as created_at, now() as updated_at, 'FLEET_REMOVE_DRIVER' as user_action_type from atlas_bpp_dashboard.role) on conflict do nothing;
\ No newline at end of file
+insert into atlas_bpp_dashboard.access_matrix (id, role_id, api_entity, user_access_type, created_at, updated_at, user_action_type) (select atlas_bpp_dashboard.uuid_generate_v4() as id,id as role_id,'FLEET' as api_entity, 'USER_FULL_ACCESS' as user_access_type, now() as created_at, now() as updated_at, 'FLEET_REMOVE_DRIVER' as user_action_type from atlas_bpp_dashboard.role) on conflict do nothing;
```
Comment (hkmangla) [inline Backend/dev/migrations/provider-dashboard/0022-fleet-owner-access.sql:6]: why FULL_ACCESS to everyone 😓

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs [hunk 2]
```diff
@@ -855,7 +856,15 @@ getFleetVehicleAssociation _merchantShortId fleetOwnerId mbLimit mbOffset = do
 
         (driverName, driverId) <- case currentActiveDriver of
           Just driver -> pure (driver.firstName <> " " <> fromMaybe "" driver.lastName, Just driver.id.getId)
-          Nothing -> pure ("", Nothing)
+          Nothing -> do
+            latestAssociation <- QRCAssociation.findLatestLinkedByRCId vrc.id now
+            case latestAssociation of
+              Just latestAssociation' -> do
+                driver <- QPerson.findById latestAssociation'.driverId
+                case driver of
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs:864]: Not good way of writing code when you have to handle Maybe with empty string. Handle it better way.

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/dev/migrations/provider-dashboard/0022-fleet-owner-access.sql (hunk 1 above)
Comment (AshishSaini25) [reply Backend/dev/migrations/provider-dashboard/0022-fleet-owner-access.sql:6]: ohh i added this is local by mistake it came here i will change.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs (hunk 2 above)
Comment (AshishSaini25) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs:864]: sure i will fix it.

**Review by hkmangla — APPROVED** [review]

**Pull request merged**